src/backend/distributed/transaction/transaction_recovery.c

```diff
@@ -225,17 +230,26 @@ RecoverWorkerTransactions(WorkerNode *workerNode)
 			appendStringInfo(command, "ROLLBACK PREPARED '%s'", transactionName);
 		}
 
-		result = PQexec(connection, command->data);
-		if (PQresultStatus(result) != PGRES_COMMAND_OK)
+		querySent = SendRemoteCommand(connection, command->data);
+		if (querySent == 0)
 		{
-			WarnRemoteError(connection, result);
+			ReportConnectionError(connection, WARNING);
+
+			break;
+		}
+
+		result = GetRemoteCommandResult(connection, raiseInterrupts);
+		if (!IsResponseOK(result))
```

Comment:
Might be useful. We should probably merge the existing ExecuteRemoteCommand and ExecuteCriticalRemoteCommand into one function and get rid of the former.